cmake/Dependencies.cmake

```diff
@@ -1,5 +1,5 @@
 TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
   LIB_REQUIRED_PACKAGES KokkosCore KokkosContainers KokkosAlgorithms 
-  LIB_OPTIONAL_TPLS quadmath MKL BLAS LAPACK CUSPARSE MAGMA
+  LIB_OPTIONAL_TPLS quadmath MKL BLAS LAPACK CUSPARSE MAGMA SUPERLU CHOLMOD LAPACKE CBLAS
```

Comment:
Not that I know of.

Reply:
We might have a bootstrap issue here for LAPACKE and CBLAS. The tpls don't exist in Trilinos (AFAIK) so adding the TPLs to `Dependencies.cmake` would also require adding Tribits find modules.

Reply:
Trilinos does have find modules for SuperLU and Cholmod, but with different capitalization.
`SUPERLU -> SuperLU` and `CHOLMOD -> Cholmod`.

This problem of capitalize vs dont' capitalize is going to follow us forever...

Reply:
@jjwilke I'm not sure how easy it would be to change these in a backwards-compatible way so that Trilinos could accept either capitalization, even for specific TPLs. @bartlettroscoe might have some ideas.

Reply:
> I'm not sure how easy it would be to change these in a backwards-compatible way so that Trilinos could accept either capitalization, even for specific TPLs.

This is an ugly problem.  The names and capitalization of these TPLs should have been considered very carefully from the beginning.  But with people breaking backward compatiblity left and right right now (because of the anticipated Trilinos 13.0 release) this is a great time to fix a bunch of these issues.

@jjwilke, can we get a list of TPLs defined in {{Trilinos/TPLsList.cmake}} that should be renamed or recapitalized to match standard CMake names for these TPLs? 

I think this is why `find_package()` can also find `<lower_case_name>_config.cmake` filkes in addition to `<UpperCaseName>Config.cmake` files as this solves the capitalization problem, at least for installed packages that provide config.cmake files.

Reply:
@bartlettroscoe wrote:
> But with people breaking backward compatiblity left and right right now (because of the anticipated Trilinos 13.0 release) this is a great time to fix a bunch of these issues.

Is there a way to add "aliases" for TPL names?  Could I set up a TPL so that users could say `TPL_ENABLE_Bar=ON` and `TPL_ENABLE_Foo=ON` gets set?

Reply:
> Is there a way to add "aliases" for TPL names? Could I set up a TPL Foo so that users could say TPL_ENABLE_Bar=ON and TPL_ENABLE_Foo=ON gets set?

@mhoemmen, not sure.  Would have to look at all of the use cases that would need to work (and it is easy to leave out use cases that don't occur to one right away).

Reply:
@bartlettroscoe The main reason would be to support changes in capitalization.  On the other hand, CMake already tells you about unused options, so perhaps that's good enough to support a change.